### PR TITLE
initiated dashboard page where user can find their wishlist, earned badges, and pathways

### PR DESCRIPTION
**Pull request by malynda**
re: Issue #57

### CODE REVIEW

**Comment by stenington** [comment]
Throwing a Work In Progress tag on this so I don't aggressively merge before ready.

**Comment by shiba4life** [comment]
@stenington we got the count working on dashboard title.

**Comment by stenington** [comment]
@chloeatplay http://discovery-189.herokuapp.com/ is up to see this PR.

**Comment by threeqube** [comment]
I got a weird "login error - audience mismatch : domain mismatch" error when trying to log in via Persona to that heroku link.

**Comment by stenington** [comment]
@threeqube Oh okay, I will fix that. Thanks!

**Comment by stenington** [comment]
@chloe @threeqube Ok you should be able to log in.

**Comment by chloeatplay** [comment]
:thumbsup:

**Comment by threeqube** [comment]
Thanks @stenington. I seem to be able to access this page when I click on my identity - sunny@mofo. Is this the intended way?

**Comment by chloeatplay** [comment]
@threeqube I think yes for now. Is this ticket ready to close? cc @mchizek

**Comment by threeqube** [comment]
@chloeatplay gotcha.

**Comment by malynda** [comment]
It's weird that the icons and counts aren't in there, I wonder what happened with that?

**Comment by shiba4life** [comment]
@mchizek I think we can merge the pull request?  @stenington just want to confirm that you've reviewed it?

**Comment by stenington** [comment]
Haven't had a chance to review it, and I'd also like to figure out why it looked different than @mchizek expected in heroku.

**Comment by stenington** [comment]
Sorry, I thought commits were missing from this PR somehow, but I just saw that there were a few more I could pull down on my machine and push to heroku. So hopefully http://discovery-189.herokuapp.com/ looks like you'd expect it to @mchizek?

**Comment by shiba4life** [comment]
@stenington looks great!  Can we merge?

File: clientapp/router.js [hunk 1]
```diff
@@ -108,6 +110,33 @@ module.exports = Backbone.Router.extend({
...
+          source: Achievements.BACKPACK
+      });
+      var wishlist = new Achievements({
+          pageSize: 4,
+          source: Achievements.WISHLIST,
+          type: Achievement.BADGE
+      });
+      var pathways = new Achievements({
+              pageSize: 4,
+              source: Achievements.PATHWAY
```
Comment (stenington) [inline clientapp/router.js:125]: This should be `type: Achievement.Pathway` with a `pathways.fetch();` below, except when I do that I get all badges in the pathways section. :/ Not entirely sure what's going on, but probably not worth bothering about too much in favor of a more robust data strategy.

File: clientapp/router.js [hunk 2]
```diff
@@ -108,6 +110,33 @@ module.exports = Backbone.Router.extend({
...
+      });
+      backpack.fetch();
+      wishlist.fetch();
+      app.renderPage(new DashboardPage({
+        model: window.app,
+        collection: this.listing,
+        addSources: {
```
Comment (stenington) [inline clientapp/router.js:132]: The option name here is a bit misleading. I used `addSources` elsewhere meaning "sources from which to add badges" but it's just an arbitrary name. Maybe it should just be called `sources` or something.

File: clientapp/templates/includes/dash-panel.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<div>
+  <dl class="accordion" data-accordion>
+    <dd>
+      <a href="#pathwayPanel">Pathways</a>
+      <div id="pathwayPanel" class="js-pathway-items content active">
```
Comment (stenington) [inline clientapp/templates/includes/dash-panel.html:5]: This div and the following 2 being active is a bit of a misuse of the accordion, as you can never get back to that state through the controls provided. I think we can just get a designer to take a look though and if they're fine with it I'm fine with it.

File: clientapp/templates/includes/dash-title.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<div>
+  <h1>{{ currentUser.email }}</h1>
+  <div class="row text-center">
+	  <div class="small-4 columns">
+	  	<span class="fa-stack fa-lg">
```
Comment (stenington) [inline clientapp/templates/includes/dash-title.html:5]: Ha, cool! Didn't know about font awesome stacking.

File: clientapp/templates/includes/dash-title.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+	  		<i class="fa fa-circle fa-stack-2x"></i>
+	  		<i class="fa fa-code-fork fa-stack-1x fa-inverse"></i>
+	  	</span>
+	  	<p>{{pathways.models.length}} pathways<br>created</p>
```
Comment (stenington) [inline clientapp/templates/includes/dash-title.html:9]: `models.length` is probably wrong here, as I don't think the collection will necessarily contain all of a user's pathways or badges or whatever at any given moment. But that's also a detail that I think can be changed later again when we have a better sense of data flow.

File: clientapp/templates/pages/dashboard.html [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  </section>
+  <section class="dashboard-data banner">
+    <div class="row text-center">
+      <div class="small-12 large-8 small-centered columns dash-title"></div>
+    </div>
+  </section>
+  <!--<section class="pathway-editor banner">-->
+  <section class="dashboard-editor banner">
+    <div class="row text-center">
+      <div class="js-stack stack">
```
Comment (stenington) [inline clientapp/templates/pages/dashboard.html:17]: `js-stack` and `stack` are unnecessary here I think. `js-stack` at least is a hook into creating layered panels that can be reordered through button controls (think of a stack in the z direction) so it's not really wrong to have it, but it's a bit misleading.

File: clientapp/views/includes/dash-title.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+var templates = require('templates');
+
+module.exports = HumanView.extend({
+  template: templates.includes.dashTitle,
+  initialize: function (opts) {
+    this.currentUser = window.app.currentUser;
+    this.backpack = opts.sources.backpack;
+    this.wishlist = opts.sources.wishlist;
+    this.pathways = opts.sources.pathways;
+    this.listenToAndRun(this.backpack, 'add', this.render);
```
Comment (stenington) [inline clientapp/views/includes/dash-title.js:11]: This is something I'm not sure about, but it seems like calling `render` from within `initialize` is a bad practice. Several of the human-model helpers take care of rendering, and in my limited experience it seems like over-eager rendering can lead to problems where you're drawing things before data is fully prepared, or trying to wedge components into other components without elements in the DOM yet, etc.

I would probably make these just `listenTo` calls though I haven't tried that to make sure the dash-title is still initially rendered if you do.

File: clientapp/views/layout.js [hunk 8]
```diff
@@ -35,5 +36,9 @@ module.exports = HumanView.extend({
     // TODO: generalize the following to all(?) dropdown buttons
     var dropdown = $(evt.currentTarget).closest('.f-dropdown');
     Foundation.libs.dropdown.close(dropdown);
+  },
+  dashboard: function (evt) {
```
Comment (stenington) [inline clientapp/views/layout.js:40]: At some point we might want to introduce a generalized link click handler for links with `data-route-to="page"` or something, but this is fine for now.

**Comment by stenington** [comment]
A couple other things I noticed:
- Usernames should look like links (cursor change or underline or something)
- `View more` buttons should hide unless there's more to view
- Clicking a pledged pathway should take you to the pledged pathway page

**Comment by stenington** [comment]
I think because we're trying to move a bit quickly it might serve us best to go ahead and merge this, and I'll make my comments into new tickets.

**Comment by chloeatplay** [comment]
:thumbsup:

**Pull request merged**